### PR TITLE
Windows, test wrapper: implement zip support

### PR DESCRIPTION
**Pull request by laszlocsomor**
The test wrapper can now create zip files. We will
use this feature to archive the test's undeclared
outputs.

See https://github.com/bazelbuild/bazel/issues/5508

Change-Id: I94f41a1cca128e5a0cfbbdc1e7c5f71248f9091a

### CODE REVIEW

File: tools/test/windows/tw.cc [hunk 1]
```diff
@@ -492,6 +550,26 @@ bool TouchFile(const std::wstring& path) {
   return true;
 }
 
+bool ReadCompleteFile(HANDLE handle, uint8_t* dest, DWORD max_read_bytes) {
+  if (max_read_bytes == 0) {
+    return true;
+  }
+
+  const DWORD max_read = std::min(
+      max_read_bytes, /* 100 MB */ static_cast<DWORD>(100000000));
```
Comment (meteorcloudy) [inline tools/test/windows/tw.cc]: What undeclared outputs are possible, is any of them going to exceed 100MB?

Comment (laszlocsomor) [reply tools/test/windows/tw.cc]: Tests are free to write anything into `$TEST_UNDECLARED_OUTPUTS_DIR`.
In theory they could dump large files there.

Comment (laszlocsomor) [reply tools/test/windows/tw.cc]: Actually, this restriction makes no sense. There's no reason to limit single ReadFile calls to 100 MB. Let me remove this code.

**Comment by laszlocsomor** [comment]
Thanks! I removed the 100MB chunk sizes limit.

**Pull request closed without merging**